skimage/feature/scattering.py

```diff
@@ -0,0 +1,286 @@
+        (N,spatial_coefs,spatial_coefs)
+    Raises
+    ------
+    UserWarning
+        If the size of x is not (N,px,px) and informs that the images with be
+        cropped to (N,px,px)
+    UserWarning
+        If no wavelet filters, the function creates a multiresolution set of
+        filters with predefined settings, but warns about the parameters and
+        suggests precomputing the filters.
+    UserWarning
+        If the value of m is not 0,1, or 2.
+    References
+    ----------
+    .. [1] Bruna, J., Mallat, S. 'Invariant Scattering Convolutional Networks'.
+    IEEE Transactions on PAMI, 2012.
+    .. [2] Oyallon, E. et Mallat, S. 'Deep Roto-translation Scattering for
+    Object Classification'. CVPR 2015.
+    Examples
+    --------
+    Processing a set of 3 (randomly generated) images:
+       >>>> import numpy as np
+       >>>> from scattering.filter_bank \
+                import multiresolution_filter_bank_morlet2d
+       >>>> from scattering.scattering import scattering
+       >>>> px = 32 #number of pixels of the images
+       >>>> images = np.arange(0,3*px*px).reshape(3,px,px) #create images
+       >>>> wavelet_filters,lp = \
+                multiresolution_filter_bank_morlet2d(px,J=np.log2(px))
+       >>>> S,U = scattering(images,m=2)
+    """
+    # Check that the input data is correct:
+    # 1.- Signals are squared, otherwise, crop
+    assert_nD(x, 3, 'x')  # check that the images are correctly stacked
+    num_signals, px, py = x.shape
+    if px != py:
+        warning_string = "Variable x has shape {0}, " \
+                         "which is not in format (N,px,px). " \
+                         "We crop to the smallest dimension."
+        warnings.warn(warning_string.format(x.shape))
+        px = min(px, py)
+        x = x[:, 0:px, 0:px]
+    # 2.- If we dont have filters, get them with the default values, J=3, L=8
+    if wavelet_filters is None:
+        J = int(min(np.log2(px), 3))
+        L = 8
+        warning_string = "No filter input, we create a " \
+                         "Morlet filter bank with " \
+                         "J= {0} and L={1}. " \
+                         "Strongly suggest creating " \
+                         "the filters before hand and" \
+                         " pass them as a parameter."
+        warnings.warn(warning_string.format(J, L))
+        wavelet_filters, lp = multiresolution_filter_bank_morlet2d(px,
+                                                                   J=J, L=L)
+    else:
+        J = len(wavelet_filters['psi'][0])  # number of scales
+        L = len(wavelet_filters['psi'][0][0])  # number of orientations
+    num_signals = x.shape[0]
+    # 3.- Check the Order (m) of the scattering transform, can only be 0,1,2,
+    # and that gives us different number of scattering coefficients
+    num_coefs = {
+        0: int(1),
+        1: int(1 + J * L),
+        2: int(1 + J * L + J * (J - 1) * L ** 2 / 2)
+    }.get(m, -1)
```

Comment:
honestly, this is more of a case selection, which in python is done with `if ... elif ... elif ... ... else`. Seems more readable to me. Then you just raise the exception in the `else` branch